api/src/chat/utils/rcdaChatDialog.ts

```diff
@@ -13,19 +16,19 @@ export default function rcdaChatDialog<TDialogData, TDependencies>(
 export function rcdaChatDialogStateful<TDialogData, TDependencies>(
```

Comment:
Just curious but what is the purpose of this function Max?

Reply:
This allows for a dialog to define a dialogData constructor. The constructor will be run once per dialog, before the first waterfall dialog step. session.dialogData for that dialog will then be strongly typed according to the result of the dialogData constructor. I may later collapse this into the normal rcdaChatDialog method as an overload.